stats/datadog/collector.go

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package datadog
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/loadimpact/k6/stats/statsd/common"
+	log "github.com/sirupsen/logrus"
+	null "gopkg.in/guregu/null.v3"
+)
+
+type tagHandler sort.StringSlice
+
+func (t tagHandler) filterTags(tags map[string]string, group string) []string {
+	var res []string
+
+	for key, value := range tags {
+		if value != "" && t.contains(key) {
+			res = append(res, key+":"+value)
+		}
+	}
+	return append(res, "group:"+group)
+}
+
+func (t tagHandler) contains(key string) bool {
+	var n = ((sort.StringSlice)(t)).Search(key)
+	return n != len(t) && t[n] == key
+}
+
+// Config defines the datadog configuration
+type Config struct {
+	common.Config
+
+	TagWhitelist null.String `json:"tag_whitelist,omitempty" envconfig:"tag_whitelist" default:"status, method"`
```

Comment:
Several notes here:
- I'd prefer if we don't use the `default` struct tag to set the defaults, since this seems like a functionality from [envconfig](https://github.com/kelseyhightower/envconfig#struct-tag-support), which we plan to replace (https://github.com/loadimpact/k6/issues/671). 
- Also, I'm pretty sure having the default values be managed by `envconfig` will mean that these values can never be configured by the JSON config. Any values there will just be overwritten... So some sort of constructor seems best for now...
- `null.String` is likely far from the best type to use here, though I admit the map type we use in k6 is also far from ideal as well... (https://github.com/loadimpact/k6/issues/755). Although, here we'll likely have to go with a map, since this can also contain user-defined tags, not only the known system tags.
- Finally, I already mentioned my issues with the chosen default values - they seem too few, and also for some reason always implicitly include the `group`, even if its not specified here.